src/psyclone/tests/dynamo0p3_transformations_test.py

```diff
@@ -6616,40 +6616,123 @@ def test_intergrid_err(dist_mem):
         lftrans.apply(loops[0], loops[1])
     assert expected_err in str(excinfo)
 
+# Start OpenACC section
 
-def test_no_acc():
-    '''
-    Check that attempting to add any sort of OpenACC directive to a
-    dynamo0p3 InvokeSchedule causes an error.
 
-    '''
-    from psyclone.transformations import ACCEnterDataTrans, ACCLoopTrans, \
-        ACCParallelTrans
-    accdt = ACCEnterDataTrans()
-    accpt = ACCParallelTrans()
-    acclt = ACCLoopTrans()
+# Class ACCEnterDataTrans start
+def test_accenterdatatrans():
+    '''Test that an ACCEnterDataTrans transformation can add an OpenACC
+    Enter Data directive to the PSy-layer in the dynamo0.3 API.
 
+    '''
+    from psyclone.transformations import ACCEnterDataTrans
+    acc_enter_trans = ACCEnterDataTrans()
     _, info = parse(os.path.join(os.path.dirname(os.path.abspath(__file__)),
                                  "test_files", "dynamo0p3",
-                                 "1_single_invoke.f90"),
-                    api=TEST_API)
+                                 "1_single_invoke.f90"))
+    psy = PSyFactory(TEST_API, distributed_memory=False).create(info)
+    sched = psy.invokes.get('invoke_0_testkern_type').schedule
+    _ = acc_enter_trans.apply(sched)
+    code = str(psy.gen)
+    assert (
+        "      !$acc enter data\n"
```

Comment:
Why doesn't this `enter data` specify what data to move to the device? The kernel must use at least one field and that should be listed unless I'm missing something? Or is it that there are no OpenACC parallel or kernels directives? In which case I think the code-generation step should raise an exception as it makes no sense to have one without the other.

Reply:
It is the latter. Enter Data looks at the ACCParallel directives within its region to determine data to move. I've added a check at code generation time.